domain_utils/domain_utils.py

```diff
@@ -116,14 +116,61 @@ def hostname_subparts(url, include_ps=False, **kwargs):
     return subparts
 
 
-def get_stripped_url(url, scheme=False):
-    """Returns a url stripped to (scheme)?+hostname+path"""
+def get_stripped_url(url, scheme=False, non_http_scheme=None):
+    """
+    Returns a url stripped to (scheme)?+netloc+path
+    For example ``https://my.domain.net/a/path/to/a/file.html#anchor?a=1``
+    becomes ``my.domain.net/a/path/to/a/file.html``
+
+
+    URL parsing is done using std lib urllib.parse.urlparse
+    Using netloc means that a port is included, for example,
+    if it was in the path.
+    The method strips just the beginning and end being stripped.
+
+    :param url: URL to be parsed
+    :type url: str
+    :param scheme: If True, scheme will be prepended in
+        returned result, defaults to False
+    :type scheme: bool, optional
+    :param non_http_scheme: Action to take if scheme is not
+        ``http`` or ``https`` e.g. ``file:`` or ``about:blank``.
+        If None, return empty string.
+        If ``self``, return the original URL.
+        Default is None.
+    :type non_http_scheme: None or ``"self"``, optional
+
+    :return: Returns a url stripped to (scheme)?+netloc+path.
+        Returns empty string if appropriate.
+    :rtype: str
+    """
+    if non_http_scheme not in [None, 'self']:
+        raise ValueError('non_http_scheme must be either `None` or `self`')
     purl = urlparse(url)
-    surl = ''
-    if scheme:
-        surl += purl.scheme + '://'
-    try:
-        surl += purl.hostname + purl.path
-    except TypeError:
-        surl += purl.hostname
-    return surl
+
+    _scheme = purl.scheme
+    scheme_out = ''
+    netloc_out = purl.netloc
+    path_out = purl.path
+
+    if _scheme not in ['http', 'https']:
+        if non_http_scheme == 'self':
+            scheme = True
+        if non_http_scheme is None:
+            # e.g. in the case of about:blank, the path is 'blank', but we want
+            # to return nothing
+            path_out = ''
```

Comment:
The docstring says `None` will return an empty string. You can just `return ''` here?

Reply:
yes. this is better. and will clarify that it's short circuiting and scheme etc. will not be considered.